### PR TITLE
 Implemented Block Status Check API Integration with UI and Tests

### PR DESCRIPTION
**Pull request by vinaysaichelupuri**
#  Implemented Block Status Check API Integration with UI and Tests

### This PR adds functionality to check the block status between two users via an API call. The following changes have been made.

- Created checkBlockStatus utility function to call the /api/users/block-status endpoint using fetch.
- Handled the response based on API status (success and error cases).
- Integrated the block status check in the UI to reflect API results dynamically.

## Test Coverage:
All edge cases for the API call have been unit tested using Jest.
Verified UI behavior with different API responses.

### CODE REVIEW

File: src/components/IndividualChatHeader/IndividualChatHeader.tsx [hunk 1]
```diff
@@ -1,17 +1,26 @@
 import {useNavigation} from '@react-navigation/native';
 import {Image, Platform, Text, TouchableOpacity, View} from 'react-native';
-
 import {useThemeColors} from '../../themes/colors';
 import {useImagesColors} from '../../themes/images';
 import {HomeStackProps} from '../../types/usenavigation.type';
 import {UserDetails} from '../../types/user.types';
+import {ChatOptionsModal} from '../ChatOptionsModal/ChatOptionsModal';
 import {individualChatHeaderStyles} from './IndividualChatHeader.styles';
+import {useState} from 'react';
```
Comment (varun-martha) [inline src/components/IndividualChatHeader/IndividualChatHeader.tsx:10]: Check order.

File: src/screens/IndividualChat/IndividualChat.test.tsx
Comment (varun-martha) [inline src/screens/IndividualChat/IndividualChat.test.tsx:1]: Remove comments.

File: src/screens/IndividualChat/IndividualChat.tsx [hunk 2]
```diff
@@ -52,11 +54,34 @@ export const IndividualChat = ({route}: Props) => {
...
+          blockerPhoneNumber: userData.phoneNumber,
+          blockedPhoneNumber: user.phoneNumber,
+          authToken: token,
+        });
+        if (result.status === 200) {
+          setIsUserBlocked(result.data.isBlocked);
+        }
+      }
+    } catch (error) {
+      console.error('Error checking block status:', error);
```
Comment (varun-martha) [inline src/screens/IndividualChat/IndividualChat.tsx:74]: Handle this error, instead of console.

**Review by varun-martha — APPROVED** [review]
Kindly do necessary changes and merge it.
